stream.go

```diff
@@ -0,0 +1,61 @@
+package manta
+
+import (
+	"io"
+
+	"github.com/dotabuff/manta/dota"
+)
+
+// stream wraps an io.Reader to provide functions necessary for reading the
+// outer replay structure.
+type stream struct {
+	r io.Reader
```

Comment:
if you embed this anonymously, the type `stream` will also be an `io.Reader`, which means anything written to operate on an `io.Reader` will work on `stream` out of box.